plugins/modules/openshift_versions.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: openshift_versions
+
+short_description: Query supported OpenShift versions
+
+version_added: "1.0.0"
+
+description: Retrieves the list of supported OpenShift versions.
+
+
+options:
+  version:
+    description: Version of OpenShift
+    required: False
+    type: str
+
+  only_latest:
+    description: Retrieve only latest minor version
+    required: False
+    default: False
+    type: bool
+
+author:
+    - Michele Costa  (@nocturnalstro)
+"""
+
+EXAMPLES = r"""
+- name: Query OpenShift architectures
+  openshift_versions:
+    openshift_version: 4.16.19
+"""
+
+RETURN = r"""
+versions:
+  description: A list of supported OpenShift versions
+  type: dict
+  returned: always
+  sample: {
+    "4.18.1": {
+        "cpu_architectures": [
+            "ppc64le",
+            "x86_64",
+            "arm64",
+            "s390x"
+        ],
+        "default": true,
+        "display_name": "4.18.1",
+        "support_level": "production"
+    },
+    "4.18.1-multi": {
+        "cpu_architectures": [
+            "x86_64",
+            "arm64",
+            "s390x",
+            "ppc64le"
+        ],
+        "display_name": "4.18.1-multi",
+        "support_level": "production"
+    }
+  }
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import missing_required_lib
+
+import traceback
+
+try:
+    from ansible_collections.openshift_lab.assisted_installer.plugins.module_utils import apitoken, apiurl
+except ImportError:
+    from ansible.module_utils import apitoken, apiurl
+
+
+try:
+    import requests
+except ImportError:
+    HAS_REQUESTS = False
+    REQUESTS_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_REQUESTS = True
+    REQUESTS_IMPORT_ERROR = None
+
+
+def run_module():
+    module_args = dict(
+        version=dict(type="str", required=False),
+        only_latest=dict(type="bool", required=False, default=False)
+    )
+
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=False)
+
+    # Fail if requests is not installed
+    if not HAS_REQUESTS:
+        module.fail_json(msg=missing_required_lib('requests'), exception=REQUESTS_IMPORT_ERROR)
+
+    # Set headers
+    headers = {
+        'Content-Type': 'application/json',
+        'Authorization': f'Bearer {apitoken.GetToken()}'
+    }
+
+    query_params = {
```

Comment:
Curious why query_params are being set individually instead of looping through them similar to other modules, e.g. https://github.com/vjayaramrh/assistedinstaller/blob/3e87e9abd381fe8b2eb1e2843abfe3554a653582/plugins/modules/support_levels.py#L154

Reply:
Its only two values seemed, simpler to just do that way :)

Reply:
In my opinion, It would make it consistent with other modules.

Reply:
Would appreciate if you can make the change, Thanks